.gitignore

```diff
@@ -16,13 +16,55 @@ testbin/
 /webrev/
 /webrev.zip
 
+# IntelliJ Idea Begin
+# https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
 # Ignore IntelliJ files
-.idea/tasks.xml
-.idea/workspace.xml
 .idea/inspectionProfiles/**
 .idea/copyright/profiles_settings.xml
-.idea/codeStyles/**
-.idea/checkstyle-idea.xml
+
+# Code styles should be shared
+#.idea/codeStyles/**
+#.idea/checkstyle-idea.xml
+
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
+
+# Generated files
+.idea/**/contentModel.xml
+
+# Gradle
+.idea/**/gradle.xml
+.idea/**/libraries
+
+# Auto-import
+.idea/artifacts
+.idea/compiler.xml
+.idea/jarRepositories.xml
+.idea/modules.xml
+.idea/*.iml
+.idea/modules
+*.iml
+*.ipr
+
+# File-based project format
+*.iws
+
+# IntelliJ
+out/
```

Comment:
This would pick up more than just IntelliJ files. Are they all needed?

Reply:
`out` is the default `Compiler output` path in IntelliJ. But I don't think it is really needed.

Reply:
Updated it, `*.iml` and `*.ipr` are the Idea project files that are created when importing from gradle.